launch/launch/actions/include_launch_description.py

```diff
@@ -160,8 +161,12 @@ def execute(self, context: LaunchContext) -> List[LaunchDescriptionEntity]:
             perform_substitutions(context, normalize_to_list_of_substitutions(arg_name))
             for arg_name, arg_value in self.launch_arguments
         ]
-        declared_launch_arguments = (
-            launch_description.get_launch_arguments_with_include_launch_description_actions())
+        try:
+            declared_launch_arguments = (
+                launch_description.get_launch_arguments_with_include_launch_description_actions())
+        except Exception as exc:
+            exc.add_note(f'while executing {self.describe()}')
```

Comment:
Unfortunately, `add_node()` was added in Python 3.11: https://docs.python.org/3/library/exceptions.html#BaseException.add_note. Windows uses Python 3.8 and RHEL 9 uses Python 3.9: https://www.ros.org/reps/rep-2000.html#jazzy-jalisco-may-2024-may-2029.

This is falling a test on RHEL (and would probably result in a nasty/confusing error at runtime): https://ci.ros2.org/job/ci_linux-rhel/2015/testReport/junit/launch.test/test_mypy/test_mypy/

```
201 files checked
2 errors
launch/actions/include_launch_description.py: note: In member "execute" of class "IncludeLaunchDescription":
launch/actions/include_launch_description.py:168:13: error: "Exception" has no attribute "add_note"
Found 1 error in 1 file (checked 201 source files)
```

You could do something like this:

```suggestion
            if hasattr(exc, 'add_note'):
                exc.add_note(f'while executing {self.describe()}')
```

Same for `launch/launch/launch_description.py`. Not sure if there's something similar to `add_note()` in Python <3.11.